flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/LocalClusteringCoefficientExample.java

```diff
@@ -0,0 +1,205 @@
+		DataSet<Tuple2<Long, HashSet<Long>>> neighborhoods = graph
+				.reduceOnEdges(new NeighborhoodEdgesFunction(), EdgeDirection.OUT);
+
+		// Construct a new graph where the neighborhood is the vertex value
+		Graph<Long, HashSet<Long>, NullValue> newGraph = graph
+				.mapVertices(new EmptyVertexMapFunction())
+				.joinWithVertices(neighborhoods, new NeighborhoodVertexMapFunction());
+
+		// Calculate clustering coefficient
+		DataSet<Tuple2<Long, Double>> clusteringCoefficients = newGraph
+				.reduceOnNeighbors(new ClusteringCoefficientNeighborsFunction(), EdgeDirection.OUT);
+
+		// Emit results
+		if(fileOutput) {
+			clusteringCoefficients.writeAsCsv(outputPath, "\n", ",");
+		} else {
+			clusteringCoefficients.print();
+		}
+
+		env.execute("Local Clustering Coefficient Example");
+	}
+
+	// --------------------------------------------------------------------------------------------
+	//  Clustering Coefficient Functions
+	// --------------------------------------------------------------------------------------------
+
+	private static final class NeighborhoodEdgesFunction
+			implements EdgesFunction<Long, NullValue, Tuple2<Long, HashSet<Long>>> {
+
+		@Override
+		public Tuple2<Long, HashSet<Long>> iterateEdges(
+				Iterable<Tuple2<Long, Edge<Long, NullValue>>> edges) throws Exception {
+
+			Long vertexId = null;
+			HashSet<Long> neighbors = new HashSet<Long>();
+
+			for (Tuple2<Long, Edge<Long, NullValue>> edge : edges) {
+				vertexId = edge.f0;
+				neighbors.add(edge.f1.getTarget());
+			}
+
+			return new Tuple2<Long, HashSet<Long>>(vertexId, neighbors);
+		}
+	}
+
+	private static final class EmptyVertexMapFunction
+			implements MapFunction<Vertex<Long, NullValue>, HashSet<Long>> {
+
+		@Override
+		public HashSet<Long> map(Vertex<Long, NullValue> arg) throws Exception {
+			return new HashSet<Long>();
+		}
+	}
+
+	private static final class NeighborhoodVertexMapFunction
+			implements MapFunction<Tuple2<HashSet<Long>, HashSet<Long>>, HashSet<Long>> {
+
+		@Override
+		public HashSet<Long> map(Tuple2<HashSet<Long>, HashSet<Long>> arg) throws Exception {
+			return arg.f1;
+		}
+	}
+
+	private static final class ClusteringCoefficientNeighborsFunction
+			implements NeighborsFunctionWithVertexValue<Long, HashSet<Long>, NullValue, Tuple2<Long, Double>> {
+
+		@Override
+		public Tuple2<Long, Double> iterateNeighbors(Vertex<Long, HashSet<Long>> vertex,
+				Iterable<Tuple2<Edge<Long, NullValue>, Vertex<Long, HashSet<Long>>>> neighbors) throws Exception {
+
+			int deg = vertex.getValue().size();
+			int e = 0;
+
+			// Calculate common neighbor count (e)
+			for (Tuple2<Edge<Long, NullValue>, Vertex<Long, HashSet<Long>>> neighbor : neighbors) {
+				// Iterate neighbor's neighbors
+				for (Long nn : neighbor.f1.getValue()) {
+					if (vertex.getValue().contains(nn)) {
+						e++;
+					}
+				}
+			}
+
+			// We assume an undirected graph, so we need to divide e here
+			e /= 2;
```

Comment:
I don't get the reasoning behind this.. Do you assume an undirected graph as input? or you convert the input to undirected? in any case, I think we should add a comment in the beginning of the example to define the behavior.